src/main/java/io/vertx/lang/jruby/JRubyVerticle.java

```diff
@@ -0,0 +1,120 @@
+package io.vertx.lang.jruby;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Verticle;
+import io.vertx.core.Vertx;
+import io.vertx.core.spi.VerticleFactory;
+import org.jruby.CompatVersion;
+import org.jruby.RubyClass;
+import org.jruby.RubyInstanceConfig;
+import org.jruby.embed.LocalContextScope;
+import org.jruby.embed.ScriptingContainer;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.net.URL;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class JRubyVerticle implements Verticle {
+
+  private final JRubyVerticleFactory factory;
+  private final ClassLoader classLoader;
+  private final String verticleName;
+  private Vertx vertx;
+  private Context context;
+  private ScriptingContainer container;
+
+  public JRubyVerticle(JRubyVerticleFactory factory, ClassLoader classLoader, String verticleName) {
+    this.factory = factory;
+    this.classLoader = classLoader;
+    this.verticleName = VerticleFactory.removePrefix(verticleName);
+  }
+
+  @Override
+  public Vertx getVertx() {
+    return vertx;
+  }
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    this.vertx = vertx;
+    this.context = context;
+  }
+
+  @Override
+  public void start(Future<Void> startFuture) throws Exception {
+    String gemPath = context.config().getString("GEM_PATH");
+    if (gemPath != null && !gemPath.trim().isEmpty()) {
+      container = factory.createContainer(gemPath);
```

Comment:
I don't think it's right for verticles to have their own container - we'll run of ouf RAM very quickly.